pkg/common/cns-lib/vsphere/vsan.go

```diff
@@ -50,3 +52,59 @@ func (vc *VirtualCenter) DisconnectVsan(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (vc *VirtualCenter) IsvSANFileServiceEnabled(ctx context.Context) (bool, error) {
+	log := logger.GetLogger(ctx)
+	datacenters, err := vc.ListDatacenters(ctx)
+	if err != nil {
+		log.Errorf("failed to find datacenters from VC: %q, Error: %+v", vc.Config.Host, err)
+		return false, err
+	}
+
+	err = vc.ConnectVsan(ctx)
+	if err != nil {
+		log.Errorf("error occurred while connecting to VSAN, err: %+v", err)
+		return false, err
+	}
+
+	// Get clusters from datacenters.
+	clusterComputeResources := []*object.ClusterComputeResource{}
+	for _, datacenter := range datacenters {
+		finder := find.NewFinder(datacenter.Datacenter.Client(), false)
+		finder.SetDatacenter(datacenter.Datacenter)
+		clusterComputeResource, err := finder.ClusterComputeResourceList(ctx, "*")
+		if err != nil {
+			if _, ok := err.(*find.NotFoundError); ok {
+				log.Errorf("No clusterComputeResource found in datacenter: %+v. error: %+v", datacenter, err)
+				continue
+			}
+			log.Errorf("Error occurred while getting clusterComputeResource. error: %+v", err)
+			return false, err
+		}
+		clusterComputeResources = append(clusterComputeResources, clusterComputeResource...)
+	}
+	isVsanFileServiceEnabled := false
+	// Check whether vSAN FS is enabled on cluster.
+	for _, cluster := range clusterComputeResources {
+		// Get the cluster config to know if file service is enabled on it or not.
+		config, err := vc.VsanClient.VsanClusterGetConfig(ctx, cluster.Reference())
+		if err != nil {
+			log.Errorf("failed to get the vsan cluster config. error: %+v", err)
+			return false, err
+		}
+		if !(*config.Enabled) {
+			log.Infof("cluster: %+v is a non-vSAN cluster. Skipping this cluster", cluster)
+			continue
+		} else if config.FileServiceConfig == nil {
+			log.Debugf("VsanClusterGetConfig.FileServiceConfig is empty. Skipping this cluster: %+v with config: %+v",
+				cluster, config)
+			continue
+		}
+		log.Debugf("cluster: %+v has vSAN file services enabled: %t", cluster, config.FileServiceConfig.Enabled)
+		if config.FileServiceConfig.Enabled {
+			isVsanFileServiceEnabled = true
```

Comment:
vSan file service gets enabled on a VC cluster level. Is it correct to set this flag true if it's enabled on any cluster in the VC?

Reply:
@divyenpatel Could you comment on Ankit's comment?